enigma/src/main/java/cuchaz/enigma/source/SourceRemapper.java

```diff
@@ -13,14 +13,21 @@ public SourceRemapper(String source, Iterable<Token> tokens) {
 	}
 
 	public Result remap(Remapper remapper) {
-		StringBuffer remappedSource = new StringBuffer(this.source);
+		StringBuilder remappedSource = new StringBuilder(this.source);
 		Map<Token, Token> remappedTokens = new HashMap<>();
 
 		int accumulatedOffset = 0;
 		for (Token token : this.tokens) {
 			Token movedToken = token.move(accumulatedOffset);
 
 			String remappedName = remapper.remap(token, movedToken);
+
+			// fixes a bug where the remapped name was displaying each and every inner class (e.g A.A.B.A.B.C instead of simply A.B.C)
+			// we do this by removing everything but the last class in the name (eg A.B.C -> C)
+			if (remappedName != null && remappedName.contains(".")) {
```

Comment:
ngl this feels a bit jank, should it not be fixed in the `Remapper` implementation?